pagefaultgames/pokerogue
# [Bug] Rockruff, lycanroc wild spawn fix

## What are the changes?
There's a few different changes here.

Change 1: Allows Own Tempo Rockruff to spawn in the wild, at a rate of 20%, so 20% of all rockruff's you might find can have Own Tempo Form, this allows for somewhat easier access to a pokemon that has the same base stats as its other forms, just put together differently.

Change 2: I changed Dusk Form Lycanroc from being form index 2, to form index 1, this makes it so if you run into a wild Dusk Form Lycanroc and haven't yet unlocked Own Tempo Rockruff, catching this Lycanroc will unlock Own Tempo Rockruff.

Change 3: I changed Lycanroc's forms from spawning based on Time of Day(including any Lycanroc owned by a enemy trainer), and instead just made it based on a % chance, the rates are 40% for Day form, 40% for Midnight form, and 20% for Dusk form.
I personally think it's silly to have the forms locked to a certain time of day in the first place, as Lycanroc itself isn't exactly a super strong pokemon to begin with.

## Why am I doing these changes?
Before, Own Tempo Rockruff couldn't spawn in the wild at all, and in order to get access to it you had to catch a midnight form Lycanroc.


### Screenshots/Videos
Midnight:
![image](https://github.com/user-attachments/assets/10932218-9a8e-44ea-8927-361297b44418)

Dusk:
![image](https://github.com/user-attachments/assets/f0f7d35d-cd42-42ac-9a7c-e711385eeb85)

Day:
![image](https://github.com/user-attachments/assets/cd83d296-f2b8-459b-9a2e-eaa207139d6d)

Own Tempo in the wild:
![image](https://github.com/user-attachments/assets/5902e91b-c786-498a-a7c2-b4d52786628b)


## How to test the changes?
Change your OPP_SPECIES_OVERRIDE to either 744 for Rockruff or 745 for Lycanroc
If on a fresh save file, you can see that catching a Dusk Lycanroc does indeed unlock Own Tempo Rockruff, while catching Midnight or Day Form only unlocks normal Rockruff.
And you can see that Lycanroc's forms can all spawn no matter the time of day, just based on random chance.


also sorry I really cannot figure out the whole cherry picking thing D:

## Checklist
